src/api/app/models/workflow.rb

```diff
@@ -35,13 +37,17 @@ def call
       return unless event_matches_event_filter?
       return unless branch_matches_branches_filter?
 
-      case
-      when scm_webhook.closed_merged_pull_request?
-        destroy_target_projects
-      when scm_webhook.reopened_pull_request?
-        restore_target_projects
-      when scm_webhook.new_pull_request?, scm_webhook.updated_pull_request?, scm_webhook.push_event?, scm_webhook.tag_push_event?
-        steps.each do |step|
+      steps.each do |step|
+        case
+        when (step.is_a? Workflow::Step::SubmitRequest)
```

Comment:
This seems brittle as it only works if this stays on top. I think this could be easier solved with moving *all* of this into each of the `Workflow::Step`s. It's their responsibility and not the one of `Workflow`.

Reply:
I think the best option would be to do this here too and do something similar as the other methods

https://github.com/openSUSE/open-build-service/blob/c739e4438e31959501e202c00197d1c26668fef9/src/api/app/models/workflow.rb#L133

Reply:
I moved this out to a follow up card now: https://trello.com/c/IeJx39EZ